src/journal-remote/journal-upload-journal.c

```diff
@@ -300,6 +327,22 @@ static size_t journal_input_callback(void *buf, size_t size, size_t nmemb, void
                           u->entries_sent, u->current_cursor);
         }
 
+        if (filled > 0 && u->compression.algorithm != COMPRESSION_NONE) {
+                size_t compressed_size;
+                r = compress_blob(u->compression.algorithm, compression_buffer, filled, buf, size * nmemb, &compressed_size, u->compression.level);
+                if (r < 0) {
+                        log_error_errno(r, "Failed to compress %zu bytes using Compression=%s, CompressionLevel=%d)",
+                                        filled, compression_lowercase_to_string(u->compression.algorithm), u->compression.level);
+                        return CURL_READFUNC_ABORT;
+                }
+                if (compressed_size > size * nmemb) {
```

Comment:
I do not think this happens. When compressed result is larger than the original, `compress_blob()` should fail. Please convert this `if` condition to `assert()`.

Reply:
i was asked [here](https://github.com/systemd/systemd/pull/34822#discussion_r1825693954) to add this check

Reply:
But how? compress_blob() should return -ENOBUFS in such case, and the error is checked in the above.

Reply:
right, replaced this with assert